src/layout/layout.rs

```diff
@@ -11,7 +11,8 @@ use lru::LruCache;
 use super::{Flex, SegmentSize};
 use crate::prelude::*;
 
-type Cache = LruCache<(Rect, Layout), Rc<[Rect]>>;
+type Rects = Rc<[Rect]>;
+type Cache = LruCache<(Rect, Layout), (Rects, Rects)>;
```

Comment:
Previously `Cache` mapped `(Rect, Layout)` to `[Rect]` but now it maps `(Rect, Layout)` to `([Rect], [Rect])`.

The first is the rects as a solution to the segments that the user provided constraints for, and the second is the rects that are a spacers.

The second number of `Rect`s is always the first's length + 1, because spacers are added around all user provided constraints.

Reply:
Would it be worth pulling this out into a small named type instead of a tuple?

Reply:
Do you mean like this?

```rust
type Segments = Rc<[Rect]>;
type Spacers = Rc<[Rect]>;
type Cache = LruCache<(Rect, Layout), (Segments, Spacers)>;
```

Reply:
Interesting, but I actually meant:
```rust
struct { segments, spacers }
```
Not a big deal, as this is internal only. :) I just don't like having to look up what the order of some tuple is when the items have distinctly useful names.